### PR TITLE
Implement buffered stream

### PR DESCRIPTION
**Pull request by wtdcode**
Add close() and closed() to SimpleStreamBase

Set seekable to false for SimpleStringBuffer

### Which kind of PR do you create?

- [x] This PR only contains minor fixes.
- [ ] This PR contains major feature update.
- [ ] This PR introduces a new function/api for Qiling Framework.

### Coding convention?

- [x] The new code conforms to Qiling Framework naming convention.
- [x] The imports are arranged properly.
- [x] Essential comments are added.
- [x] The reference of the new code is pointed out.

### Extra tests?

- [x] No extra tests are needed for this PR.
- [ ] I have added enough tests for this PR.
- [ ] Tests will be added after some discussion and review.

### Changelog?

- [x] This PR doesn't need to update Changelog.
- [ ] Changelog will be updated after some proper review.
- [ ] Changelog has been updated in my PR.

### Target branch?

- [x] The target branch is dev branch.

### One last thing

- [x] I have read the [contribution guide](https://docs.qiling.io/en/latest/contribution/)

-----

This implements `SimpleBufferedStream` and adds some methods for `SimpleStreamBase` and `SimpleStringBuffer`.

### CODE REVIEW

**Review by xwings — APPROVED** [review]

File: qiling/extensions/pipe.py [hunk 1]
```diff
@@ -84,3 +95,70 @@ def flush(self) -> None:
...
+    def readline(self, limit: int = -1) -> bytes:
+        ret = bytearray()
+
+        while not (ret.endswith(b'\n') or len(ret) == limit):
+            ret.extend(self.read(1))
+
+        return bytes(ret)
+
+    def write(self, s: bytes) -> int:
+        self.buff.extend(s)
```
Comment (elicn) [inline qiling/extensions/pipe.py:150]: Writing to a buffered stream should overwrite the content starting from current position.

File: qiling/extensions/pipe.py [hunk 2]
```diff
@@ -84,3 +95,70 @@ def flush(self) -> None:
...
+
+    def seek(self, offset: int, origin: int) -> int:
+        return self.lseek(offset, origin)
+    
+    def tell(self) -> int:
+        return self.cur
+
+    def read(self, n: int = -1) -> bytes:
+        if n == -1:
+            ret = self.buff
```
Comment (elicn) [inline qiling/extensions/pipe.py:130]: `n == -1` means "read till the end of the buffer".
I think it should start reading from the current position.

File: qiling/extensions/pipe.py [hunk 3]
```diff
@@ -2,6 +2,7 @@
 #
 # Cross Platform and Multi Architecture Advanced Binary Emulation Framework
 #
+import os
```
Comment (elicn) [inline qiling/extensions/pipe.py:5]: Why is it needed?

**Review by elicn — CHANGES_REQUESTED** [review]

File: qiling/extensions/pipe.py (hunk 1 above)
Comment (wtdcode) [reply qiling/extensions/pipe.py]: Done

File: qiling/extensions/pipe.py (hunk 2 above)
Comment (wtdcode) [reply qiling/extensions/pipe.py]: Done

File: qiling/extensions/pipe.py (hunk 3 above)
Comment (wtdcode) [reply qiling/extensions/pipe.py]: Removed

**Comment by wtdcode** [comment]
Closed due to #1075

**Pull request closed without merging**